backend/executor.go

```diff
@@ -121,12 +124,15 @@ func (executor *grpcExecutor) ExecuteOrchestrator(ctx context.Context, iid api.I
 	}
 
 	// Wait for the connected worker to signal that it's done executing the work-item
-	// TODO: Timeout logic - i.e. handle the case where we never hear back from the remote worker (due to a hang, etc.).
 	select {
 	case <-ctx.Done():
 		executor.logger.Warnf("%s: context canceled before receiving orchestrator result", iid)
 		return nil, ctx.Err()
 	case <-result.complete:
+		executor.logger.Debug("orchestrator", iid, "got result")
```

Comment:
nit: I'd prefer we follow the existing logging convention. Happy to change this convention if there's a better one as part of a separate PR.

```suggestion
		executor.logger.Debug("%s: orchestrator got result", iid)
```